src/mds/Server.cc

```diff
@@ -475,6 +475,12 @@ void Server::handle_client_reclaim(const cref_t<MClientReclaim> &m)
     return;
   }
 
+  std::string_view fs_name = mds->get_fs_name();
+  if (!fs_name.empty() && !session->fs_name_capable(fs_name, MAY_READ)) {
+    dout(0) << " dropping message not allowed for this fs_name: " << *m << dendl;
+    return;
+  }
```

Comment:
Why do we need this check?

Reply:
On reading this piece it made sense but I'll get back after this part of code.